cli/compose/types/types.go

```diff
@@ -62,6 +65,74 @@ type ConfigDetails struct {
 	Environment map[string]string
 }
 
+// Duration is a thin wrapper around time.Duration with improved JSON marshalling
+type Duration struct {
+	time.Duration
+}
+
+// MarshalJSON makes Duration implement json.Marshaler
+func (d Duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// UnmarshalJSON makes Duration implement json.Unmarshaler
+func (d *Duration) UnmarshalJSON(b []byte) error {
```

Comment:
MarshalJSON/Yaml needs to exist in this package to serialize, but I'm not sure UnmarshalJSON/YAML need to exist. I see there was a comment about, but I think I'm missing some context. Wouldn't it be better to keep using `mapstructure` hooks to decode?

Reply:
@dnephin Thanks! I needed those originally because I hadn't tinkered with the `mapstructure` hooks, but they're obsolete now. I've removed them as a result.